bin/ocamlformat-rpc/main.ml

```diff
@@ -0,0 +1,172 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                              OCamlFormat                               *)
+(*                                                                        *)
+(*            Copyright (c) Facebook, Inc. and its affiliates.            *)
+(*                                                                        *)
+(*      This source code is licensed under the MIT license found in       *)
+(*      the LICENSE file in the root directory of this source tree.       *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** OCamlFormat-RPC *)
+
+open Ocamlformat_lib
+
+;;
+Caml.at_exit (Format.pp_print_flush Format.err_formatter)
+
+;;
+Caml.at_exit (Format_.pp_print_flush Format_.err_formatter)
+
+module V = struct
+  type t = V1
+
+  let is_handled = function "v1" | "V1" -> Some V1 | _ -> None
+
+  let propose_another = function "v1" | "V1" -> None | _ -> Some V1
+
+  let to_string = function V1 -> "v1"
+end
+
+type state = Waiting_for_version | Version_defined of (V.t * Conf.t)
+
+let format fg conf source =
+  let input_name = "<rpc input>" in
+  let opts =
+    Conf.{debug= false; margin_check= false; format_invalid_files= false}
+  in
+  Translation_unit.parse_and_format fg ~input_name ~source conf opts
+
+let rec rpc_main = function
+  | Waiting_for_version -> (
+    match Ocamlformat_rpc_lib.Init.read_input stdin with
+    | `Halt -> Ok ()
+    | `Unknown -> Ok ()
+    | `Version vstr -> (
+      match V.is_handled vstr with
+      | Some v ->
+          Ocamlformat_rpc_lib.Init.output stdout (`Version vstr) ;
+          Out_channel.flush stdout ;
+          rpc_main (Version_defined (v, Conf.default_profile))
+      | None -> (
+        match V.propose_another vstr with
+        | Some v ->
+            let vstr = V.to_string v in
+            Ocamlformat_rpc_lib.Init.output stdout (`Version vstr) ;
+            rpc_main Waiting_for_version
+        | None -> Ok () ) ) )
```

Comment:
What does this `Ok` mean ? I suggest merging `is_handled` and `propose_another`: 
```ocaml
val handshake : string -> [ `Handled of t | `Propose_another of t ]
```

Reply:
This means that we have no other version to propose (just ending the main loop), but this should not happen since we should always suggest at least v1 indeed. that's a better handshake function indeed!